client/Hooks/useLogger/Hook.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+import { useEffect, useRef } from 'react';
+
+const MESSAGE_BUFFER_MAX_SIZE = 100;
+// TODO: Replace with the regex from the LOGGING query param
+const defaultLoggingRegex = /.*/;
+// TODO: Replace with the web socket address from config
+const defaultWebSocketAddress = 'ws://localhost:3000/log';
+const defaultClientID = 'defaultClientID';
+
+type LogLevel = 'fatal' | 'error' | 'warn' | 'info' | 'debug' | 'trace';
+
+interface loggerMessage {
+  clientTime: number;
+  clientID: string;
+  clientLevel: LogLevel;
+  componentName: string;
+  msg: string;
+}
+
+export const useLogger: (string) => (LogLevel, string) => void = (
+  componentName: string
+) => {
+  // TODO: store the logger websocket in context, so you get one for the whole UI, rather than one every time you call useLogger.
```

Comment:
Feels like we need this before we can merge?

Reply:
Agreed, and the same for the other TODOs (https://github.com/strimzi/strimzi-ui/pull/85/files#diff-3e52b502269a41a2c06ce466fd7d975b726b3557bc98a54d84b74b2bf958f1baR9-R12)

Reply:
Is this still blocking merge? Feels like yes?

Reply:
I think it does. Opening a websocket for every log request defeats the purpose of using a websocket.

Reply:
Latest commit resolves this via context